src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { ReactNode, useEffect } from "react";
+import ReactDOM from "react-dom";
+
+type ModalProps = {
+  setShowModal?: React.Dispatch<React.SetStateAction<boolean>>;
+  children: ReactNode;
+  closeOnEscape?: boolean;
+};
+
+const modalRoot = document.getElementById("modal-root");
+
+const Modal: React.FC<ModalProps> = ({
+  setShowModal,
+  children,
+  closeOnEscape = false,
+}: ModalProps) => {
+  const handleKeyPress = (e: KeyboardEvent) => {
```

Comment:
`useCallback` probably. 
It will make sure the function won't be recreated everytime